Core/Graphics/Source/Graphics.cpp

```diff
@@ -2,13 +2,15 @@
 
 #include <JsRuntimeInternalState.h>
 
+#include <cassert>
+
 #define BGFX_RESET_FLAGS (BGFX_RESET_VSYNC | BGFX_RESET_MSAA_X4 | BGFX_RESET_MAXANISOTROPY)
 
 namespace Babylon
 {
     namespace
     {
-        constexpr auto JS_SENTINEL_NAME = "graphicsInitializationPromise";
+        constexpr auto JS_GRAPHICS_READY_NAME = "whenGraphicsReady";
```

Comment:
I don't mind changing the name, though we already use the "when" prefix for task/promise returning functions in many places in both the native and JS code, so I don't think it is out of place. Do you have suggestions for a different name (that is still consistent with other parts of the code)?

As far as a promise returning function vs an event/callback, I strongly prefer a promise returning function. We need to await this state inside initialization code, and having to do it through state+event is both harder for the consumer and more error prone. For example, you need expose the current state and the caller needs to check it before registering for the event. If you forget to check the state, then your event handler will never trigger, and your code basically hangs. This could easily happen depending on the order you call the various initialization functions in Babylon Native. Additionally, even if you got this code right, you'd still have to "convert" the state+event into a promise that you can await in your init code. All of that seems way more complicated for the consumer than just awaiting a promise returning function.

Reply:
We need to agree on the contract for initialization in general. We can decide on the naming then.

Reply:
Do we want to address that in this PR, or separately? I think the name here is perfectly workable, so if we want to check it in now and address the initialization contract in a separate issue/meeting/PR, that sounds reasonable to me.